nibabel/filebasedimages.py

```diff
@@ -0,0 +1,569 @@
+                      'use ``to_filename`` instead',
+                      DeprecationWarning, stacklevel=2)
+        self.to_filename(filename)
+
+    def to_file_map(self, file_map=None):
+        raise NotImplementedError
+
+    def to_files(self, file_map=None):
+        warnings.warn('``to_files`` method is deprecated\n'
+                      'Please use the ``to_file_map`` method '
+                      'instead',
+                      DeprecationWarning, stacklevel=2)
+        self.to_file_map(file_map)
+
+    @classmethod
+    def make_file_map(klass, mapping=None):
+        ''' Class method to make files holder for this image type
+
+        Parameters
+        ----------
+        mapping : None or mapping, optional
+           mapping with keys corresponding to image file types (such as
+           'image', 'header' etc, depending on image class) and values
+           that are filenames or file-like.  Default is None
+
+        Returns
+        -------
+        file_map : dict
+           dict with string keys given by first entry in tuples in
+           sequence klass.files_types, and values of type FileHolder,
+           where FileHolder objects have default values, other than
+           those given by `mapping`
+        '''
+        if mapping is None:
+            mapping = {}
+        file_map = {}
+        for key, ext in klass.files_types:
+            file_map[key] = FileHolder()
+            mapval = mapping.get(key, None)
+            if isinstance(mapval, basestring):
+                file_map[key].filename = mapval
+            elif hasattr(mapval, 'tell'):
+                file_map[key].fileobj = mapval
+        return file_map
+
+    load = from_filename
+
+    @classmethod
+    def instance_to_filename(klass, img, filename):
+        ''' Save `img` in our own format, to name implied by `filename`
+
+        This is a class method
+
+        Parameters
+        ----------
+        img : ``spatialimage`` instance
+           In fact, an object with the API of ``spatialimage`` - specifically
+           ``dataobj``, ``affine``, ``header`` and ``extra``.
+        filename : str
+           Filename, implying name to which to save image.
+        '''
+        img = klass.from_image(img)
+        img.to_filename(filename)
+
+    @classmethod
+    def from_image(klass, img):
+        ''' Class method to create new instance of own class from `img`
+
+        Parameters
+        ----------
+        img : ``spatialimage`` instance
+           In fact, an object with the API of ``spatialimage`` -
+           specifically ``dataobj``, ``affine``, ``header`` and ``extra``.
+
+        Returns
+        -------
+        cimg : ``spatialimage`` instance
+           Image, of our own class
+        '''
+        return klass(img.dataobj,
+                     img.affine,
+                     klass.header_class.from_header(img.header),
+                     extra=img.extra.copy())
```

Comment:
No. Anything with `dataobj` or `affine` should not appear in `filebasedimage`. There are also some dirty docs; fixing this now.